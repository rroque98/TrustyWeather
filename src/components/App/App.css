:root {
  --mainDarkText: #3d3d3f;
  --mainLight: #f5f5f5;
  --mainbackgroundDark: #1f7bd1;
  --backgroundForecastItems: #155a9b;
  --whitish: #f9f9f9;
  --borderColor: #ccd1cb;
}

body {
  background: var(--mainLight);
  padding: 1em;
}

.appWrapper {
  display: grid;
  grid-gap: 10px;
}

.appTitle {
  background: var(--mainbackgroundDark);
  grid-column: 1 / 3;
  grid-row: 1 / 2;
  font-weight: 800;
  font-size: 2rem;
  color: var(--whitish);
  padding: 1rem;
}

.locationContainer {
  background: var(--whitish);
  grid-column: 1 / 3;
  grid-row: 3 / 4;
  padding: 1em;
  font-weight: 500;
  font-size: 1.5em;
  color: var(--mainDarkText);
}
