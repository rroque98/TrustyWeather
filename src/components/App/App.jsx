import React, { Component } from "react";
import axios from "axios";
import Search from "../Search";
import CurrentWeather from "../CurrentWeather";
import ForecastList from "../ForecastList";
import SubmitButton from "../SubmitButton";

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      location: "oakland",
      searchValue: ""
    };
    this.getWeatherData = this.getWeatherData.bind(this);
    this.updateSearchValue = this.updateSearchValue.bind(this);
    this.handleSearchClick = this.handleSearchClick.bind(this);
  }

  componentDidMount() {
    this.getWeatherData();
  }

  getWeatherData() {
    const proxyurl = "https://cors-anywhere.herokuapp.com/";
    const endpoint = `https://www.metaweather.com/api/location/search/?query=${this.state.location}`;
    axios
      .get(proxyurl + endpoint)
      .then(locationData => {
        let locationWoeid = locationData.data[0].woeid;
        let weatherEndpoint = `https://www.metaweather.com/api/location/${locationWoeid}/`;
        axios
          .get(proxyurl + weatherEndpoint)
          .then(weatherData => {
            this.setState({
              currWeather: weatherData.data.consolidated_weather[0],
              fiveDayForecast: weatherData.data.consolidated_weather.slice(1)
            });
          })
          .catch(err => {
            console.log(err);
          });
      })
      .catch(err => {
        console.log(err);
      });
  }

  updateSearchValue(e) {
    this.setState({
      searchValue: e.target.value
    });
  }

  handleSearchClick(e) {
    e.preventDefault();
    this.setState({ location: this.state.searchValue }, this.getWeatherData);
  }

  render() {
    return (
      <div className="appContainer">
        <header>Trusty Weather</header>
        <Search
          updateSearchValue={this.updateSearchValue}
          searchValue={this.state.searchValue}
          placholder="Search by city name"
        />
        <SubmitButton value="Search" handleClick={this.handleSearchClick} />
        <CurrentWeather currWeather={this.state.currWeather} />
        <ForecastList fiveDayForecast={this.state.fiveDayForecast} />
      </div>
    );
  }
}

export default App;
