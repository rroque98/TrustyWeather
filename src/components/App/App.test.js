import React from "react";
import { render, cleanup } from "@testing-library/react";
import App from "./index.js";

afterEach(cleanup);

it("renders", () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});
