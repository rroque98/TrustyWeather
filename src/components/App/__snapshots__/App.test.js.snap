// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="appWrapper"
  >
    <header
      class="appTitle"
    >
      Trusty Weather
    </header>
    <header
      class="locationContainer"
    />
    <section>
      <input
        class="searchBar"
        placeholder="Search by city name"
        type="search"
        value=""
      />
    </section>
    <div>
      <input
        type="submit"
        value="Search"
      />
    </div>
    <article
      class="forecastListContainer"
    />
  </div>
</DocumentFragment>
`;
